browser/resources/settings/email_aliases_page/stories/index.tsx

```diff
@@ -0,0 +1,240 @@
+  aliases: [
+    {
+      email: 'horse.radish.record57@bravealias.com',
+      note: 'Alias for all my newsletters',
+      domains: undefined
+    },
+    {
+      email: 'start.plane.division.laser42@bravealias.com',
+      domains: ['x.com'],
+      note: 'Twitter account'
+    },
+    {
+      email: 'racoon.pencil.test14@bravealias.com',
+      note: 'Marketplace email for Facebook',
+      domains: undefined
+    }
+  ]
+} satisfies { email: string, aliases: Alias[] }
+
+class MockEmailAliasesService implements EmailAliasesServiceInterface {
+  accountEmail: string
+  aliases: Map<string, Alias>
+  accountState: AccountState
+  accountRequestId: number
+  observers: Set<EmailAliasesServiceObserverRemote |
+                 EmailAliasesServiceObserverInterface>
+  constructor(accountState: AccountState, accountEmail: string) {
+    this.accountState = accountState
+    this.accountEmail = accountEmail
+    this.observers = new Set<EmailAliasesServiceObserverRemote |
+                             EmailAliasesServiceObserverInterface>()
+    this.aliases = new Map<string, Alias>();
+    for (const alias of demoData.aliases) {
+      this.aliases.set(alias.email, alias)
+    }
+  }
+
+  addObserver (observer: EmailAliasesServiceObserverRemote |
+                         EmailAliasesServiceObserverInterface) {
+    this.observers.add(observer)
+    switch (this.accountState) {
+      case 'NoAccount':
+        observer.onLoggedOut()
+        break
+      case 'AccountReady':
+        observer.onLoggedIn(this.accountEmail)
+        break
+      case 'AwaitingAccount':
+        observer.onVerificationPending(this.accountEmail)
+        break
+    }
+    observer.onAliasesUpdated([...this.aliases.values()])
+  }
+
+  removeObserver (observer: EmailAliasesServiceObserverRemote |
+                            EmailAliasesServiceObserverInterface) {
+    this.observers.delete(observer)
+  }
+
+  updateAlias (email: string, note: string) {
+    const alias = { email, note, domains: undefined }
+    this.aliases.set(email, alias)
+    this.observers.forEach(observer => {
+      observer.onAliasesUpdated([...this.aliases.values()])
+    })
+  }
+
+  deleteAlias (email: string) {
+    this.aliases.delete(email)
+    this.observers.forEach(observer => {
+      observer.onAliasesUpdated([...this.aliases.values()])
+    })
+  }
+
+  async generateAlias () {
+    let generated: string = ''
+    do {
+      generated = "mock-" + Math.random().toString().slice(2,6) +
+        "@bravealias.com"
+    } while (this.aliases.has(generated))
+    await new Promise(resolve => setTimeout(resolve, 1000))
+    return { alias: generated }
+  }
+
+  requestPrimaryEmailVerification (email: string) {
+    this.observers.forEach(observer => {
+      observer.onVerificationPending(email)
+    })
+    this.accountRequestId = window.setTimeout(() => {
+      this.observers.forEach(observer => {
+        observer.onLoggedIn(email)
+      })
+    }, 5000);
```

Comment:
what are you using this timeout for? We shouldn't be doing this in unit tests as they should designed to be fast

Reply:
This is not part of the unit tests. It's the storybook mock backend that allows us to manually observe the behavior.

Reply:
I guess it's a bit confusing here because this is called `MockEmailAliasesService` and the one used in tests is called `MockEmailAliasesServiceInterface`.

Reply:
I would change this one to `StubEmailAliasesService` and clearly comment that it's just a temporary stub until the cpp implementation is added

Reply:
it was extra confusing because it's easy to scroll down and miss that this is in a different file than the tests. Stupid github :)

Reply:
renamed the storybook one to `StubEmailAliasesService`